src/main/java/org/folio/rest/impl/InitEventBus.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.rest.impl;
+
+import org.folio.orders.events.handlers.MessageAddress;
+import org.folio.rest.resource.interfaces.PostDeployVerticle;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+/**
+ * The class initializes event bus handlers
+ */
+public class InitEventBus implements PostDeployVerticle {
+  private final Logger logger = LoggerFactory.getLogger(InitEventBus.class);
+
+  @Autowired
+  @Qualifier("orderStatusHandler")
+  Handler<Message<JsonObject>> orderStatusHandler;
+
+  public InitEventBus() {
+    SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
+  }
+
+  @Override
+  public void init(Vertx vertx, Context context, Handler<AsyncResult<Boolean>> resultHandler) {
+    vertx.executeBlocking(
+      handler -> {
+        EventBus eb = vertx.eventBus();
+        eb.consumer(MessageAddress.ORDER_STATUS.address, orderStatusHandler);
+
+        handler.complete();
```

Comment:
Isn't it better to complete the handler with the result from the eventbus registration?

Reply:
This is just to register consumer and a handler which is going to be used when any message is sent to the address. I.e. the handler will be called only and as many times when the message is sent. This is why `handler.complete()` is called as standalone operation just to let vertx know that all consumers created (for now it is just one but might be more defined later).

Reply:
Correct, I understand that, but I was just wondering of a scenario where the consumer registration could be delayed but we are going ahead and completing the operation. It usually happens in a clustered event bus but probably the scenario is rare or doesn't even occur in our case, but just a thought.

Reply:
Now I see what you meant. I applied change so now `localConsumer` is used just to not propagate the consumer to cluster (if deployed in such mode). Also I added logic to complete event bus init logic only when consumer is registered. In case in future this requires changes, it can be easily done here.